pilot/pkg/model/sidecar.go

```diff
@@ -828,6 +844,39 @@ func serviceMatchingVirtualServicePorts(service *Service, vsDestPorts sets.Set[i
 	return nil
 }
 
+// computeWildcardHostVirtualServiceIndex computes the wildcardHostVirtualServiceIndex for a given
+// list of virtualServices. This is used to optimize the lookup of the most specific wildcard host
+func computeWildcardHostVirtualServiceIndex(virtualServices []config.Config, services []*Service) map[host.Name]types.NamespacedName {
+	fqdnVirtualServiceHostIndex := make(map[host.Name]config.Config, len(virtualServices))
+	wildcardVirtualServiceHostIndex := make(map[host.Name]config.Config, len(virtualServices))
+	for _, vs := range virtualServices {
+		v := vs.Spec.(*networking.VirtualService)
+		for _, h := range v.Hosts {
+			if host.Name(h).IsWildCarded() {
+				wildcardVirtualServiceHostIndex[host.Name(h)] = vs
+			} else {
+				fqdnVirtualServiceHostIndex[host.Name(h)] = vs
```

Comment:
I think this is a bug; we don't check for duplicate fqdn hosts so we'd silently override it. This didn't get released so i'll put up a PR to fix it

/cc @hzxuzhonghu